src/Stackify/Log/Entities/Api/WebRequestDetail.php

```diff
@@ -93,19 +130,71 @@ private function __construct()
         $this->RequestUrl = $this->getRequestUrl();
         $this->RequestUrlRoot = filter_input(INPUT_SERVER, 'SERVER_NAME');
         $this->ReferralUrl = filter_input(INPUT_SERVER, 'HTTP_REFERER');
-        $this->Headers = $this->getHeaders();
-        $this->Cookies = isset($_COOKIE) ? self::getRequestMap($_COOKIE, true) : null;
-        $this->QueryString = isset($_GET) ? self::getRequestMap($_GET) : null;
-        $this->PostData = isset($_POST) ? self::getRequestMap($_POST) : null;
-        $this->SessionData = isset($_SESSION) ? self::getRequestMap($_SESSION, true) : null;
-        $this->PostDataRaw = file_get_contents('php://input');
+
+        /**
+         * @var \Stackify\Log\Transport\Config\Agent
+         */
+        $agentConfig = Agent::getInstance();
+
+        if ($agentConfig) {
+            $this->Headers = $agentConfig->getCaptureErrorHeaders()
+                ? $this->getHeaders(
+                    $agentConfig->getCaptureErrorHeadersBlacklist(),
+                    $agentConfig->getCaptureErrorHeadersWhitelist()
+                )
+                : null;
+            $this->Cookies = isset($_COOKIE) && $agentConfig->getCaptureErrorCookies()
+                ? self::getRequestMap(
+                    $_COOKIE,
+                    $agentConfig->getCaptureErrorCookiesBlacklist(),
+                    $agentConfig->getCaptureErrorCookiesWhitelist()
+                )
+                : null;
+
+            $this->QueryString = isset($_GET) && $agentConfig->getCaptureGetVariables()
+                ? self::getRequestMap(
+                    $_GET,
+                    $agentConfig->getCaptureGetVariablesBlacklist(),
+                    $agentConfig->getCaptureGetVariablesWhitelist()
+                )
+                : null;
+            $this->PostData = isset($_POST) && $agentConfig->getCapturePostVariables()
+                ? self::getRequestMap(
+                    $_POST,
+                    $agentConfig->getCapturePostVariablesBlacklist(),
+                    $agentConfig->getCapturePostVariablesWhitelist()
+                )
+                : null;
+            $this->SessionData = isset($_SESSION) && $agentConfig->getCaptureSessionVariables()
+                ? self::getRequestMap(
+                    $_SESSION,
+                    $agentConfig->getCaptureSessionVariablesBlacklist(),
+                    $agentConfig->getCaptureSessionVariablesWhitelist()
+                )
+                : null;
+            $this->PostDataRaw = $agentConfig->getCaptureRawPostData() ? file_get_contents('php://input'): null;
+        } else {
+            $this->Headers = $this->getHeaders();
+            $this->Cookies = isset($_COOKIE) ? self::getRequestMap($_COOKIE, ['*']) : null;
+            $this->QueryString = isset($_GET) ? self::getRequestMap($_GET) : null;
```

Comment:
Since the whitelist is null here the default would be capture none of the query string parameters and none of the POST data.  Not sure this code will ever get called, but if it does is that what we want to do?